ast/rego_v1.go

```diff
@@ -116,11 +118,45 @@ func checkDeprecatedBuiltinsForCurrentVersion(node interface{}) Errors {
 	return checkDeprecatedBuiltins(deprecatedBuiltins, node)
 }
 
-// CheckRegoV1 checks the given module for errors that are specific to Rego v1
-func CheckRegoV1(module *Module) Errors {
+// CheckRegoV1 checks the given module or rule for errors that are specific to Rego v1.
+// If something other than an *ast.Rule or *ast.Module is passed, it will be ignored and result in no errors.
```

Comment:
Currently, no other types have v1 restrictions imposed on them.
But given that this would be a programming error, we could panic for unsupported cases. WDYT?

Reply:
Yeah that sounds ok to me.